src/util/mw-api.ts

```diff
@@ -6,64 +6,55 @@ import { articleDetailXId, redirectsXId } from '../stores';
 
 export async function getArticlesByIds(_articleIds: string[], downloader: Downloader, log = true): Promise<void> {
     let from = 0;
-    let numArticleIds = _articleIds.length;
+    const numArticleIds = _articleIds.length;
     let numThumbnails = 0;
-    let batchSize = 30;
+    const maxBatchSize = 50;
 
     // using async iterator to spawn workers
     await pmap(
         ','.repeat(downloader.speed).split(',').map((_, i) => i),
         async (workerId: number) => {
             while (from < numArticleIds) {
-                const articleIds = _articleIds.slice(from, from + batchSize);
+                const articleIds = _articleIds.slice(from, from + maxBatchSize);
+                let articlesStringLength = articleIds.join('|').length;
+                while (articlesStringLength > 7900) {
+                    articlesStringLength -= encodeURI(articleIds.slice(-1)[0]).length;
+                    articleIds.pop();
```

Comment:
considering that `pop()` returns the value, you could use the value and merge these two lines.

Reply:
Why using `encodeURI` and not two line above with the `join()`?